### PR TITLE
Bump DMD to 2.109.0

### PR DESCRIPTION
**Pull request by BNAndras**

### CODE REVIEW

**Comment by BNAndras** [comment]
I  went for the external repo to simplify things. Alpine has 2.104 so I went with Ubuntu 24.04 to get 2.109.0. There's an official download script available on https://dlang.org/download.html, but we'd still need to get a copy of DUB. There are releases from GitHub we can source like https://github.com/dlang/dub/releases/tag/v1.38.0.

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-FROM dlanguage/dmd:2.080.0
-
-RUN apt-get update && \
-    apt-get install -y jq && \
-    apt-get purge --auto-remove -y && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
+FROM ubuntu:24.04
+RuN apt update
```
Comment (ErikSchierboom) [inline Dockerfile:2]: You could combine all the apt commands in a single layer using && and \
This reduces the image size

File: bin/run-tests-in-docker.sh [hunk 2]
```diff
@@ -23,6 +23,7 @@ docker run \
     --read-only \
     --mount type=bind,src="${PWD}/tests",dst=/opt/test-runner/tests \
     --mount type=tmpfs,dst=/tmp \
+    --mount type=tmpfs,dst=/root/.dub \
```
Comment (ErikSchierboom) [inline bin/run-tests-in-docker.sh:26]: This won't work in production as this flag won't be passed. Would not passing the readonly flag fix it? If so, feel free to remove that flag and see if that fixes it, as we're technically not yet passing that flag in production.

**Review by ErikSchierboom — CHANGES_REQUESTED** [review]

File: bin/run-tests-in-docker.sh (hunk 2 above)
Comment (BNAndras) [reply bin/run-tests-in-docker.sh:26]: I was getting an error about that folder being a read only location even if i made the whole image writable.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**